src/storage/archive_index.rs

```diff
@@ -0,0 +1,114 @@
+use crate::error::Result;
+use crate::storage::{compression::CompressionAlgorithm, FileRange};
+use failure::ResultExt;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::io;
+use std::path::{Path, PathBuf};
+
+#[derive(Deserialize, Serialize)]
+pub struct FileInfo {
```

Comment:
`pub` means we don't get warnings about dead code. It would be great to use `pub(crate)` instead everywhere you have pub in this file.
```suggestion
pub(crate) struct FileInfo {
```

Reply:
I updated the file (I hope to the extend you meant)